src/switchImplIntf.ml

```diff
@@ -0,0 +1,66 @@
+open Import
+
+let switch client document : string array Promise.t =
+  LanguageClient.sendRequest client ~meth:"ocamllsp/switchImplIntf"
+    ~data:(Uri.toString document.TextDocument.uri ~skipEncoding:true ())
+    ()
+
+let requestSwitch ~client ~capabilities document =
+  (* given a file uri, opens the file if it exists;
+     otherwise, creates the file but doesn't write it to disk *)
+  let showFile targetFileName =
+    let open Promise.O in
+    let uri = Uri.parse targetFileName in
+    Vscode.Workspace.openTextDocument (`Uri uri)
+    |> Js.Promise.catch (fun _ ->
+           (* if file does not exist *)
+           let createFileUri =
+             Uri.with_ uri @@ Uri.make_change ~scheme:"untitled" ()
+           in
+           Vscode.Workspace.openTextDocument (`Uri createFileUri))
+    >>= fun doc ->
+    Vscode.Window.showTextDocument (`Document doc) >>| fun _ -> ()
+  in
+
+  let filesToSwitchTo =
+    match (client, capabilities) with
+    | Some client, Some capabilities
+      when OcamlLsp.handleSwitchImplIntf capabilities ->
+      switch client document
+    | _ -> failwith "NOT IMPLEMENTED"
+    (* TODO handle more gracefully *)
```

Comment:
Are we going to keep this unimplemented?

Reply:
I think so unless failwith causes hard crash for the user.

Reply:
another option is to remind the user to upgrade ocamllsp with a possibility to "Do not show this again" but I won't be able to get to work on this until the weekend (I would really prefer to have this merged :)))

Reply:
Okay, but we can't release until we hide errors like this. I don't think we necessarily need "a do not show this again" message, but every error message should be clear and make it obvious for how to fix the problem.